backend/modules/publishing/service.py

```
@@ -1 +1,610 @@
+            return
+        
+        # Upload from base64
+        if file_contents_base64:
+            pdf_bytes = base64.b64decode(file_contents_base64)
+            s3_url = await s3_service.upload_from_bytes(pdf_bytes, s3_key, file_type)
+            
+            if s3_url:
+                self.repository.create_or_update_simulation_file(
+                    simulation.id, filename, s3_url, file_size, file_type
+                )
+                setattr(simulation, 'case_study_url', s3_url)
```

Ruff (0.14.8)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)